[[ Bug 21532 ]] Move dragboard rebind to mousedown

montegoulding opened the pull request:
This patch moves the rebinding of the dragboard at the start of a drag
to mouseDown as mouseDown is documented as a handler that can set the
dragData.

runrevmark commented on engine/src/desktop.cpp:
```diff
@@ -353,7 +353,11 @@ void MCPlatformHandleMouseMove(MCPlatformWindowRef p_window, MCPoint p_location)
 {
-	if (((MCScreenDC *)MCscreen) -> isbackdrop(p_window))
+    // ensure the dragboard is bound correctly
+    MCAutoRefcounted<MCRawClipboard> t_dragboard(MCRawClipboard::CreateSystemDragboard());
```
I wonder if this should be further down... Around line 394 - I think that's where the 'reset for potential drag' happens...

runrevmark replied:
Indeed - perhaps instead of MCdragboard->Clear()?

montegoulding replied:
hehe.. tbh I didn't look that much at the rest of `MCPlatformHandleMouseDown` once I moved it there and things worked. Yes it should replace that of course!

montegoulding committed:
[[ Bug 21532 ]] Move dragboard rebind to mousedown

This patch moves the rebinding of the dragboard at the start of a drag
to mouseDown as mouseDown is documented as a handler that can set the
dragData.

runrevmark commented:
@livecode-vulcan review ok 8d560ea

livecode-vulcan commented:
💙 review by @runrevmark ok 8d560eac7da364e3d7e4cc434c210ccaae622c32

livecode-vulcan commented:
😎 test success 8d560eac7da364e3d7e4cc434c210ccaae622c32

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/74)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/66)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/66)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/66)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/66)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1588)
* try-community-universal-ios-iphoneos11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.4/builds/115)
* try-community-universal-ios-iphonesimulator11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.4/builds/115)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1618)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1612)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1608)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1645)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1585)
